### ISSUE TITLE
Bind mounting with user Docker doesn't work as expect in /usr/share/docker/plugins

### ISSUE THREAD

**Issue by nathanleclaire** [rancher/os#467]
Hi, perhaps this is expected behavior, but when I run (container for [sshfs volumes plugin](github.com/vieux/docker-volume-sshfs)):

```console
$ docker run -d -v /usr/share/docker/plugins:/usr/share/docker/plugins nathanleclaire/docker-volumes-sshfs
```

On RancherOS, the socket does not show up on the host:

```console
$ ls /usr/share/docker/plugins
ls: /usr/share/docker/plugins: No such file or directory
```

But the file is in the container just fine:

```console
$ docker exec -it $(docker ps -lq) ls /usr/share/docker/plugins
sshfs.sock
```

It works on boot2docker legacy though.  Maybe it is an issue with userland vs. sysland docker stepping on each others' toes?  I can Gist the daemon logs (system and user) if you want, but they don't seem to contain anything interesting / relevant to this issue.

**Comment by imikushin** [comment]
@nathanleclaire That's because the filesystem tree you're seeing in console is a twisted mix of bind-mounts. Also, when you log-in to RancherOS, you're really in the console container, which can see only a few of the "real" directories bind-mounted into it (check https://github.com/rancher/os/blob/v0.3.3/os-config.yml for details). 

Your container has successfully bind-mounted a dir, so the files which it puts there will be visible from other containers which bind-mount the same dir. `console` doesn't mount `/usr/share`, so you can't see `/usr/share/docker`. 

After launching a `nathanleclaire/docker-volumes-sshfs` container, try this:
```
rancher@ivan-ros-033-n1-1:~$ docker run --rm -it -v /usr/share/docker/plugins:/usr/share/docker/plugins debian:jessie    

Unable to find image 'debian:jessie' locally
jessie: Pulling from debian
902b87aaaec9: Already exists 
9a61b6b1315e: Already exists 
Digest: sha256:dc3296472bcec6d392a2f6151a4f928c59127e996c1adce448aebc1cdd169423
Status: Downloaded newer image for debian:jessie

root@da497fc09692:/# ls -lAh /usr/share/docker/plugins/           

total 0
srw-rw---- 1 root root 0 Aug 11 07:57 sshfs.sock
```

**Comment by ibuildthecloud** [comment]
@nathanleclaire Docker and the console run in different mount namespaces.  This confuses people so much.  In RancherOS v0.4 (which we are looking to release in the next two weeks) is going to run docker and the console in the same mount namespace.  This will make this issue go away.

**Comment by nathanleclaire** [comment]
> @nathanleclaire Docker and the console run in different mount namespaces. This confuses people so much. In RancherOS v0.4 (which we are looking to release in the next two weeks) is going to run docker and the console in the same mount namespace. This will make this issue go away.

Nice.

**Comment by pwFoo** [comment]
Could you provide an example how to bind mount a binary file to console / docker from a custom system-docker container? I use RancherOS v0.4.0-rc10 at the moment.

I try to extend RancherOS with system-docker containers to provide convoy plugin or other services. But the binary /usr/bin/convoy should also be usable from RancherOS console to manage volumes.

**Comment by ibuildthecloud** [comment]
This works as expected in 0.4.

@pwFoo I have an new project called wonka coming that will make this easier :)  But... for the time being... It's not so simple to bind mount a file FROM a container into another container.  You need to create a bind mount between them and copy the file to a shared place.  But then you are copying the file.  What is better is to just create an alias or script to do

   system-docker run --rm -v /var/run/convoy/rancher/:/var/run/convoy/convoy.sock myimage/convoy /usr/bin/convoy "$@"

**Comment by pwFoo** [comment]
Thanks @ibuildthecloud,

I created and mounted /usr/local/bin/ (#604) from RancherOS to my convoy container and moved the file to the volume, but it wasn't there in the RancherOS directory. So I think it should be dependent on the RancherOS volumes used in user-docker, console, ...

```
system-docker run --rm -v /var/run/convoy/rancher/:/var/run/convoy/convoy.sock myimage/convoy /usr/bin/convoy "$@"
```

Where should I place such a script persistent and executed on reboot? I tried to build services like that as a system-docker container and tried to start it before user-docker.
